src/datasources/balances-api/balances-api.manager.ts

```diff
@@ -40,11 +43,31 @@ export class BalancesApiManager implements IBalancesApiManager {
     this.zerionBalancesApi = zerionBalancesApi;
   }
 
-  async getBalancesApi(chainId: string): Promise<IBalancesApi> {
+  async getBalancesApi(
+    chainId: string,
+    safeAddress: `0x${string}`,
+  ): Promise<IBalancesApi> {
     if (this.zerionChainIds.includes(chainId)) {
       return this.zerionBalancesApi;
     }
 
+    try {
+      const transactionApi =
+        await this.transactionApiManager.getTransactionApi(chainId);
+      await transactionApi.getSafe(safeAddress);
+      return this._getSafeBalancesApi(chainId);
+    } catch (err) {
```

Comment:
Nit: I'd we're not logging it, I'd remove the unused var. (I'm surprised the linter didn't pick this up.)

```suggestion
    } catch {
```